File: IronMan/backend/src/models/ShopModel.js


const mongoose = require('mongoose');
const ShopSchema = mongoose.Schema({
    Name:{
        type:String,
        //required:true,
    },
    Services:{
        type:Array,
        //required:true,
    },
    Cost:{
        type:Object,
        //required:true,
    },
    Area:{
        type:String,
        //required:true,
    },
});
const ShopModel = mongoose.model("Shops",ShopSchema);
module.exports = ShopModel;

File: IronMan/ironman/lib/models/shopmodel.dart
// ignore_for_file: non_constant_identifier_names

/*class Shopmodel {
  Shopmodel({
    required this.Name,
    required this.Area,
    required this.Services,
    required this.Cost,
  });
  String Name, Area;
  List<String> Services;
  Map<String, int> Cost;
}
*/
class Shopmodel {
  Shopmodel({
    required this.Name,
    required this.Area,
    required this.Services,
    required this.Cost,
  });

  String Name;
  String Area;
  List<String> Services;
  Map<String, Map<String, dynamic>> Cost;

  factory Shopmodel.fromJson(Map<String, dynamic> json) {
    return Shopmodel(
      Name: json['Name'] as String,
      Area: json['Area'] as String,
      Services: List<String>.from(json['Services']),
      Cost: Map<String, Map<String, dynamic>>.from(json['Cost']),
    );
  }
}
